sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/HistogramNumeric.scala

```diff
@@ -46,7 +47,10 @@ import org.apache.spark.sql.util.NumericHistogram
       smaller datasets. Note that this function creates a histogram with non-uniform
       bin widths. It offers no guarantees in terms of the mean-squared-error of the
       histogram, but in practice is comparable to the histograms produced by the R/S-Plus
-      statistical computing packages.
+      statistical computing packages. Note: if
+      SQLConf.spark.sql.legacy.histogramNumericPropagateInputType is set to true, the output type
```

Comment:
Let's not mention the legacy config as it's internal and only for users who have problems migrating to new Spark versions. Let's just document the new behavior.

Reply:
Done.